### PR TITLE
ucblogo: init at 6.2.2

### PR DESCRIPTION
**Pull request by hllizi**
###### Description of changes

As I couldn't find any working Logo interpreter in the nixpkgs I tried to build ucblogo. This was very unproblematic to do and I thought I might just turn it into a package.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -16024,4 +16024,10 @@
     github = "ziguana";
     githubId = 45833444;
   };
+  hllizi = {
+    name = "David Lahm";
+    email = "david@klinger-lahm.de";
+    github = "hllizi";
+    githubId = 35627697;
+  };
```
Comment (pbsds) [inline maintainers/maintainer-list.nix:16032]: Please move this into a separate commit titled `maintainers: add hllizi`

File: pkgs/development/interpreters/ucblogo/default.nix [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+, autoreconfHook
+, autoconf-archive
+, texinfo
+, tetex
+, stdenv
+, fetchFromGitHub
+, gtk3
+, lib
+}:
+stdenv.mkDerivation {
```
Comment (pbsds) [inline pkgs/development/interpreters/ucblogo/default.nix:12]:
```suggestion
}:

stdenv.mkDerivation {
```

File: pkgs/development/interpreters/ucblogo/default.nix [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  pname = "ucblogo";
+  version = "6.2.2";
+  src = fetchFromGitHub {
```
Comment (pbsds) [inline pkgs/development/interpreters/ucblogo/default.nix:15]:
```suggestion
  version = "6.2.2";

  src = fetchFromGitHub {
```

File: pkgs/development/interpreters/ucblogo/default.nix [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    rev = "0ba6ad1629f94a89f98e2bf884c1388c6ee78b6d";
+    sha256 = "FyyNUUQ0cc6ZVcIzZPctMBws1uWSnWKvYJTN0BOL/28=";
+    owner = "jrincayc";
+    repo = "ucblogo-code";
+  };
+  nativeBuildInputs = [
```
Comment (pbsds) [inline pkgs/development/interpreters/ucblogo/default.nix:21]:
```suggestion
  };

  nativeBuildInputs = [
```

File: pkgs/development/interpreters/ucblogo/default.nix [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    autoreconfHook
+    texinfo
+    tetex
+    autoconf-archive
+  ];
+  buildInputs = [
```
Comment (pbsds) [inline pkgs/development/interpreters/ucblogo/default.nix:27]:
```suggestion
  ];

  buildInputs = [
```

File: pkgs/development/interpreters/ucblogo/default.nix [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    gtk3
+    wxGTK32
+    wrapGAppsHook
+  ];
```
Comment (pbsds) [inline pkgs/development/interpreters/ucblogo/default.nix:31]:
```suggestion
  ];

```

File: pkgs/development/interpreters/ucblogo/default.nix [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+  meta = with lib; {
+    description = "Berkeley Logo interpreter";
+    maintainers = with maintainers; [ hllizi ];
+    license = lib.licenses.gpl3;
+    platforms = lib.platforms.linux;
+    homepage = "https://people.eecs.berkeley.edu/~bh/logo.html";
+  };
```
Comment (pbsds) [inline pkgs/development/interpreters/ucblogo/default.nix:38]: Please order the meta attributes as shown [here](https://github.com/jtojnar/nixpkgs-hammering/blob/main/explanations/attribute-ordering.md).

File: pkgs/development/interpreters/ucblogo/default.nix [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+    tetex
+    autoconf-archive
+  ];
+  buildInputs = [
+    gtk3
+    wxGTK32
+    wrapGAppsHook
+  ];
+  meta = with lib; {
+    description = "Berkeley Logo interpreter";
```
Comment (pbsds) [inline pkgs/development/interpreters/ucblogo/default.nix:33]:
```suggestion
    description = "Interpreter for the Berkeley Logo programming language";
```

File: pkgs/development/interpreters/ucblogo/default.nix [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+    maintainers = with maintainers; [ hllizi ];
+    license = lib.licenses.gpl3;
```
Comment (pbsds) [inline pkgs/development/interpreters/ucblogo/default.nix:35]: `gpl3` is deprecated, as it is unclear. Is it `gpl3Plus` or `gpl3Only`? [more info](https://github.com/jtojnar/nixpkgs-hammering/blob/main/explanations/unclear-gpl.md)

File: pkgs/top-level/all-packages.nix [hunk 10]
```diff
@@ -38152,4 +38152,6 @@ with pkgs;
   aitrack = libsForQt5.callPackage ../applications/misc/aitrack { };
 
   widevine-cdm = callPackage ../applications/networking/browsers/misc/widevine-cdm.nix { };
+
+  ucblogo = callPackage ../development/interpreters/ucblogo { };
```
Comment (pbsds) [inline pkgs/top-level/all-packages.nix:38156]: Putting it on the bottom is nearly guaranteed to result in a merge conflict. I reccomend moving this attribute to a line somewhere in the middle of the file.

**Review by pbsds — CHANGES_REQUESTED** [review]
Great work so far!

In addition to the comments below, please fix the current merge conflict by rebasing this pull request on top of master, and squash all the ucblogo related commits into a single `init at` commit. Finally you can push the changes using `git push --force-with-lease` 

Here are two decent guides: 
* https://garrytrinder.github.io/2020/03/squashing-commits-using-interactive-rebase
* https://www.sitepoint.com/git-interactive-rebase-guide/

**Comment by Artturin** [comment]
Author hasn't responded to reviews.

Should target staging.

Hopefully closing this gets rid of https://github.com/hllizi/nixpkgs/pull/1 which shows up in staging commits
image

File: pkgs/development/interpreters/ucblogo/default.nix [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+  nativeBuildInputs = [
+    autoreconfHook
+    texinfo
+    tetex
+    autoconf-archive
+  ];
+  buildInputs = [
+    gtk3
+    wxGTK32
+    wrapGAppsHook
```
Comment (Artturin) [inline pkgs/development/interpreters/ucblogo/default.nix:30]: should be in nativeBuildInputs
